scripts/top-search-queries/get-top-search-results.go

```diff
@@ -0,0 +1,243 @@
+}
+
+// callSearchAPI calls the live Search API using the supplied values of query and nlpWeighting for the relevant query
+// parameters. It specifies a limit of 10 results to be returned in the query response.
+func callSearchAPI(ctx context.Context, query, nlpWeighting, apiURL string) []models.Item {
+	urlStr := fmt.Sprintf("%s/search?q=%s&limit=10&nlp_weighting=%s", apiURL, query, nlpWeighting)
+	logData := log.Data{"query to construct: ": urlStr}
+	log.Info(ctx, "using SDK to call Search API", logData)
+
+	queryVals := url.Values{}
+	queryVals.Add("q", query)
+	queryVals.Add("limit", "10")
+	queryVals.Add("nlp_weighting", nlpWeighting)
+	options := sdk.Options{
+		Query: queryVals,
+	}
+	logData = log.Data{"query values: ": queryVals}
+	searchAPIClient := sdk.New(apiURL)
+	response, err := searchAPIClient.GetSearch(ctx, options)
+	if err != nil {
+		log.Fatal(ctx, "failed calling Search API", err, logData)
+	}
+
+	return response.Items
+}
+
+// readQueriesFromFile opens the text file named "search-queries.txt", in the local directory, and reads it into a
+// []string object, which it returns.
+func readQueriesFromFile(ctx context.Context, cfg runConfig) (listOfQueries []string, err error) {
+	logData := log.Data{"File name: ": cfg.InputFileName}
+	file, err := os.Open(cfg.InputFileName)
+	if err != nil {
+		log.Fatal(ctx, "failed opening file", err, logData)
+	}
+	bufferedReader := bufio.NewReader(file)
+
+	var queryString string
+	for {
+		line, err := bufferedReader.ReadString('\n')
+		if err != nil {
+			if err == io.EOF {
+				// add query from last line, of input file, unless it's empty
+				queryString = strings.TrimSpace(line)
+				if queryString == "" {
+					log.Info(ctx, "ignoring last line of input file as it's empty")
+					break
+				}
+				queryString = url.QueryEscape(queryString)
+				listOfQueries = append(listOfQueries, queryString)
+				break
+			}
+			log.Fatal(ctx, "error while reading file - failed reading row of text", err, logData)
+			break
+		}
+		// add query to list
+		queryString = strings.TrimSpace(line)
+		queryString = url.QueryEscape(queryString)
+		listOfQueries = append(listOfQueries, queryString)
+	}
+	if err = file.Close(); err != nil {
+		log.Fatal(ctx, "failed closing file", err, logData)
+	}
+	logData = log.Data{"list of queries: ": listOfQueries}
+	log.Info(ctx, "take in list of top queries", logData)
+	return listOfQueries, err
+}
+
+// writeHeaderRow writes a header row to the supplied CSV file, which will be used for the output.
+func writeHeaderRow(csvFile *os.File, ctx context.Context) {
+	headerRow := make([]string, 10)
+	headerRow[0] = "NLP on or off?"
+	headerRow[1] = "Date and time of request"
+	headerRow[2] = "Query supplied"
+	headerRow[3] = "Date of release"
+	headerRow[4] = "Type of release"
+	headerRow[5] = "Position in results"
+	headerRow[6] = "Title of release"
+	headerRow[7] = "URI of release"
+	headerRow[8] = "Edition of release"
+	headerRow[9] = "Summary of release"
```

Comment:
Another slice literal maybe?
```suggestion
	headerRow := []string{
	    "NLP on or off?"
	    "Date and time of request"
	    "Query supplied"
	    "Date of release"
	    "Type of release"
	    "Position in results"
	    "Title of release"
	    "URI of release"
	    "Edition of release"
	    "Summary of release"
	}
```